### PR TITLE
Add batch secret retrieval API support

### LINKED ISSUE #126: Add batch secret retrieval support
## Summary

Add support for the Conjur [Batch Secret Retrieval API](https://docs.cyberark.com/conjur-open-source/latest/en/content/developer/conjur_api_batch_retrieve.htm) to the Java SDK.

## Motivation

Currently the Java SDK only supports retrieving secrets one at a time via `retrieveSecret(String variableId)`. Applications that need multiple secrets must make N separate HTTP calls, which is inefficient at scale. The Conjur REST API already supports batch retrieval, and the other SDKs (Python, Ruby, Go) expose it.

## Proposed Changes

- **ResourceProvider**: Add `retrieveBatchSecrets(String... variableIds)` to the interface
- **ResourceClient**: Implement batch retrieval with proper URL encoding and JSON response parsing
- **Variables / Conjur**: Expose the batch method through the existing delegation chain
- **Endpoints**: Refactor to use `(applianceUrl, account)` as primary inputs so all service URIs are derived cleanly. Add `getBatchSecretsUri()`.
- **Tests**: Comprehensive unit tests using Mockito (26 tests covering encoding, error codes, edge cases)

## API Example

```java
Conjur conjur = new Conjur();
Map<String, String> secrets = conjur.retrieveBatchSecrets(
    "prod/db/password",
    "prod/db/username",
    "prod/api/key"
);
```

## Implementation

I have a working implementation with full test coverage ready to submit as a PR.

### PR DESCRIPTION
**Pull request by miewest**
## Summary

Adds support for the Conjur [Batch Secret Retrieval API](https://docs.cyberark.com/conjur-open-source/latest/en/content/developer/conjur_api_batch_retrieve.htm) (`GET /secrets?variable_ids=...`) to the Java SDK.

Closes #126

### New: Batch Secret Retrieval
- **`ResourceProvider`**: Added `retrieveBatchSecrets(String... variableIds)` as a default interface method
- **`ResourceClient`**: Full implementation — builds the `variable_ids` query parameter with proper percent-encoding (slashes, `@`, `+`, `&`, spaces as `%20`), parses the JSON map response via Gson, and strips the `{account}:variable:` prefix from response keys
- **`Variables`** / **`Conjur`**: Expose the batch method through the existing delegation chain

### Refactored: Endpoints
- Primary constructor is now `Endpoints(String applianceUrl, String account)` — all service URIs are derived from these two values
- Added `Endpoints(String applianceUrl, String account, String authnUrl)` for custom authenticators (LDAP, OIDC, etc.)
- Added `getBatchSecretsUri()` returning `{applianceUrl}/secrets`
- Factory methods `fromSystemProperties()` and `fromCredentials()` simplified to thin wrappers
- Removed redundant `getServiceUri()` static helper that re-read system properties

### Tests
- 26 new unit tests in `ResourceClientTest` using Mockito to mock the JAX-RS client stack
- Covers: single/multi variable batch, URL encoding edge cases (slashes, `@`, `+`, `&`, spaces), all error codes (401/403/404/422), null/empty args, deeply nested paths, response order preservation, `buildBatchQueryParam` helper, single secret retrieval, and Endpoints URI derivation

## API Usage

```java
Conjur conjur = new Conjur();

// Fetch multiple secrets in one HTTP call
Map<String, String> secrets = conjur.retrieveBatchSecrets(
    "prod/db/password",
    "prod/db/username",
    "prod/api/key"
);

String dbPass = secrets.get("prod/db/password");
```

## Testing

- All 27 unit tests pass (`mvn test`)
- Manually verified against a live Conjur server (single retrieval, batch retrieval, consistency check, 404 handling)

### CODE REVIEW

**Comment by miewest** [comment]
Ignore close, I mixed up tabs and closed the wrong PR.

**Comment by miewest** [comment]
I also created a [manual full integration test of this code](https://gist.github.com/miewest/dd277a4b66066da5271c5f895cb7e0d0) but wasn't sure if it fit in the repo. I can add it if wanted.

**Comment by szh** [comment]
Thank you for the PR! We've added it to our backlog to review.

Tracking internally as CNJR-13117.

File: src/main/java/com/cyberark/conjur/api/Endpoints.java [hunk 1]
```diff
@@ -7,61 +7,100 @@
...
  */
 public class Endpoints implements Serializable {
 
     private static final long serialVersionUID = 1L;
+    private final String applianceUrl;
+    private final String account;
     private final URI authnUri;
     private final URI secretsUri;
 
-    public Endpoints(final URI authnUri, final URI secretsUri){
```
Comment (dondany) [inline src/main/java/com/cyberark/conjur/api/Endpoints.java:18]: Removing `public` constructors from `Endpoints` is technically a breaking change for anyone who instantiated it directly. Since `Endpoints` is `public`, consumers had no guardrail telling them not to use it. 
Even though `Endpoints` is not the documented primary 
API, it is a `public` class in a published package — so any external consumer instantiating it directly(for whatever reason) will get a compile error after upgrading.

The old constructors should stay, adjusted for the new functionality.

File: src/main/java/com/cyberark/conjur/api/Endpoints.java [hunk 2]
```diff
@@ -7,61 +7,100 @@
...
+     * Create Endpoints from appliance URL and account, using standard authentication.
+     *
+     * @param applianceUrl the base Conjur appliance URL (e.g. {@code https://conjur.example.com})
+     * @param account      the Conjur account name (e.g. {@code conjur} for SaaS, or your org name)
+     */
+    public Endpoints(String applianceUrl, String account) {
+        this(applianceUrl, account, applianceUrl + "/authn");
     }
 
-    public Endpoints(String authnUri, String secretsUri){
```
Comment (dondany) [inline src/main/java/com/cyberark/conjur/api/Endpoints.java:23]: Same thing as with the other constructor.

File: src/main/java/com/cyberark/conjur/api/Conjur.java [hunk 3]
```diff
@@ -105,4 +106,14 @@ public Conjur(Token token, SSLContext sslContext) {
...
         return variables;
     }
+
+    /**
+     * Fetch multiple secret values in one invocation.
+     *
+     * @param variableIds the variable IDs to retrieve
+     * @return a map of variable ID to secret value
+     */
+    public Map<String, String> retrieveBatchSecrets(String... variableIds) {
```
Comment (dondany) [inline src/main/java/com/cyberark/conjur/api/Conjur.java:116]: This seems like an unnecessary shortcut. This method is already defined in the `variables` and should be accessed through there.

**Review by dondany — CHANGES_REQUESTED** [review]

**Comment by dondany** [comment]
Closing as it was handled internally and is available in [v3.2.0](https://github.com/cyberark/conjur-api-java/releases/tag/v3.2.0) .
Thanks for your contribution it has been included in the merged changes.
